core/src/main/java/org/kiwix/kiwixmobile/core/extensions/ViewExtensions.kt

```diff
@@ -51,9 +53,36 @@ fun View.snack(
   ).apply {
     actionStringId?.let { setAction(it) { actionClick?.invoke() } }
     setActionTextColor(actionTextColor)
+    anchorView = anchor
+    addCallback(object : BaseTransientBottomBar.BaseCallback<Snackbar>() {
+      override fun onDismissed(transientBottomBar: Snackbar?, event: Int) {
+        transientBottomBar?.anchorView = null
+      }
+    })
   }.show()
 }
 
+@SuppressLint("ShowToast")
+fun View.snack(
```

Comment:
Why do we make a new extension function? we already have this function we should modify it instead of creating a new one.
https://github.com/kiwix/kiwix-android/blob/42098a891a9e2658c560528eebe620ffdab5849e/core/src/main/java/org/kiwix/kiwixmobile/core/extensions/ViewExtensions.kt#L28

You should make the anchor parameter nullable so it won't affect other use cases of this function.

Reply:
yes that would be better , on it